src/Files.App/Views/MainPage.xaml.cs

```diff
@@ -487,6 +489,54 @@ private void RootGrid_PreviewKeyDown(object sender, KeyRoutedEventArgs e)
 			}
 		}
 
+		private bool lockFlag = false;
+		//private string[] dropableArchiveTypes = { "zip", "rar", "7z", "tar" };
+		private async void HorizontalMultitaskingControlAddButton_Drop(object sender, DragEventArgs e)
+		{
+			if (lockFlag || !FilesystemHelpers.HasDraggedStorageItems(e.DataView))
+				return;
+			lockFlag = true;
+
+			var items = (await FilesystemHelpers.GetDraggedStorageItems(e.DataView))
+				.Where(x => x.ItemType is FilesystemItemType.Directory
+				//|| dropableArchiveTypes.Contains(x.Name.Split('.').Last().ToLower())
+				);
+
+			var deferral = e.GetDeferral();
+			try {
+				foreach (var item in items)
+					await NavigationHelpers.OpenPathInNewTab(item.Path);
+				deferral.Complete(); 
+			} catch { } 
+			lockFlag = false;
+		}
+
+		private async void HorizontalMultitaskingControlAddButton_DragOver(object sender, DragEventArgs e)
+		{
+			if (!FilesystemHelpers.HasDraggedStorageItems(e.DataView))
+			{
+				e.AcceptedOperation = DataPackageOperation.None;
+				return;
+			}
+
+			bool hasValidDraggedItems = 
+				(await FilesystemHelpers.GetDraggedStorageItems(e.DataView)).Any(x => x.ItemType is FilesystemItemType.Directory
+				//|| dropableArchiveTypes.Contains(x.Name.Split('.').Last().ToLower())
+				);
+
+			if (!hasValidDraggedItems) { e.AcceptedOperation = DataPackageOperation.None; return; }
+
```

Comment:
Can you split the code onto multiple lines to match the conventions used in the rest of the code base?

Reply:
Thought it got fixed when I pressed commit suggestion, oops